src/reconcile/reconcile.h

```diff
@@ -15,11 +15,12 @@
 #define WT_REC_CHECKPOINT_RUNNING 0x008u
 #define WT_REC_CLEAN_AFTER_REC 0x010u
 #define WT_REC_EVICT 0x020u
-#define WT_REC_HS 0x040u
-#define WT_REC_IN_MEMORY 0x080u
-#define WT_REC_SCRUB 0x100u
-#define WT_REC_VISIBILITY_ERR 0x200u
-#define WT_REC_VISIBLE_ALL 0x400u
+#define WT_REC_EVICT_CALL_CLOSING 0x040u
```

Comment:
I feel this PR change is a lot of code changes just to retain a single assert. We are using this newly added macro only in one code flow. After going through this code change, I am thinking a flag-save approach could be a simple solution. I agree that it may not be a good solution, but for an assert, that could be enough.

@bvpvamsikrishna opinions?